utils/ServiceLayerUtils.js

```diff
@@ -46,6 +46,10 @@ const ServiceLayerUtils = {
         } catch(e) {
             infoFormats = ['text/plain'];
         }
+        // prefer text/html
+        if(infoFormats.includes('text/html')) {
+            infoFormats = ['text/html'];
+        }
```

Comment:
I don't know yet how to modify the QWC2 code to accomodate your use case, but the above change doesn't look correct to me. Here `infoFormats` contains all the GetFeatureInfo formats supported by the WMS service, and I don't think it makes sense to limit it to `text/html` when `text/html` is one of the supported formats. This would break existing applications.

Reply:
You're right. Also this might break if the WMS supports text/html but there is nothing configured (i.e. mapserver template).